### PR TITLE
Login: Fixes Unauthorized message showing when on login page or snapshot page

### LINKED ISSUE #33764: Unauthorized message appears before logging into Grafana
**What happened**:
The "Unauthorized" message popup in Login page

I have check that in Chrome's Dev Tools -> Network, it shows "webpack:///./public/app/core/services/echo/backends/PerformanceBackend.ts" is trying to send frontend-metrics to back-end

**What you expected to happen**:
No frontend-metrics should send to backend 

**How to reproduce it (as minimally and precisely as possible)**:
Go to Grafana login page and wait for dozens of seconds

**Anything else we need to know?**:
Docker image grafana/grafana:7.5.4 doesn't have this issue.

<details>
<summary>webpack:///./public/app/core/services/echo/backends/PerformanceBackend.ts from Dev Tools</summary>

```
import { EchoBackend, EchoEvent, EchoEventType } from '@grafana/runtime';
import { backendSrv } from '../../backend_srv';

export interface PerformanceEventPayload {
  name: string;
  value: number;
}

export interface PerformanceEvent extends EchoEvent<EchoEventType.Performance, PerformanceEventPayload> {}

export interface PerformanceBackendOptions {
  url?: string;
}

/**
 * Echo's performance metrics consumer
 * Reports performance metrics to given url (TODO)
 */
export class PerformanceBackend implements EchoBackend<PerformanceEvent, PerformanceBackendOptions> {
  private buffer: PerformanceEventPayload[] = [];
  supportedEvents = [EchoEventType.Performance];

  constructor(public options: PerformanceBackendOptions) {}

  addEvent = (e: EchoEvent) => {
    this.buffer.push(e.payload);
  };

  flush = () => {
    if (this.buffer.length === 0) {
      return;
    }

    // Currently we don't have an API for sending the metrics hence logging to console in dev environment
    if (process.env.NODE_ENV === 'development') {
      console.log('PerformanceBackend flushing:', this.buffer);
    }

    console.log('performance', this.buffer);

    backendSrv.post('/api/frontend-metrics', {
      events: this.buffer,
    });

    this.buffer = [];
  };
}
```
</details>

<details>
<summary>Log from Chrome's console</summary>

```
DevTools failed to load SourceMap: Could not load content for chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/sourcemaps/disable-console.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
DevTools failed to load SourceMap: Could not load content for chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/sourcemaps/contentscript.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
DevTools failed to load SourceMap: Could not load content for chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/sourcemaps/inpage.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
login:1 The keyword 'inner-spin-button' specified to an 'appearance' property is not standardized. It will be removed in the future.
PerformanceBackend.ts:39 performance (3) [{…}, {…}, {…}]0: {name: "dcl", value: 1011}1: {name: "frontend_boot_first-paint_time_seconds", value: 0.45}2: {name: "frontend_boot_load_time_seconds", value: 1.104}length: 3__proto__: Array(0)
fetch.js:45 POST http://grafana:8080/api/frontend-metrics 401 (Unauthorized)
(anonymous) @ fetch.js:45
e._trySubscribe @ Observable.js:43
e.subscribe @ Observable.js:29
e.call @ mergeMap.js:27
e.subscribe @ Observable.js:24
t.connect @ ConnectableObservable.js:34
e.call @ refCount.js:19
e.subscribe @ Observable.js:24
e.call @ filter.js:15
e.subscribe @ Observable.js:24
e.call @ tap.js:18
e.subscribe @ Observable.js:24
c @ innerSubscribe.js:89
t._innerSub @ mergeMap.js:72
t._tryNext @ mergeMap.js:66
t._next @ mergeMap.js:49
t.next @ Subscriber.js:53
(anonymous) @ subscribeToArray.js:5
e._trySubscribe @ Observable.js:43
e.subscribe @ Observable.js:29
e.call @ mergeMap.js:27
e.subscribe @ Observable.js:24
e.call @ catchError.js:16
e.subscribe @ Observable.js:24
e.call @ takeUntil.js:16
e.subscribe @ Observable.js:24
e.call @ throwIfEmpty.js:18
e.subscribe @ Observable.js:24
(anonymous) @ backend_srv.ts:85
t.__tryOrUnsub @ Subscriber.js:192
t.next @ Subscriber.js:130
t._next @ Subscriber.js:76
t.next @ Subscriber.js:53
t._next @ filter.js:38
t.next @ Subscriber.js:53
t.next @ Subject.js:47
(anonymous) @ ResponseQueue.ts:30
t.__tryOrUnsub @ Subscriber.js:192
t.next @ Subscriber.js:130
t._next @ Subscriber.js:76
t.next @ Subscriber.js:53
t.next @ Subject.js:47
add @ ResponseQueue.ts:35
(anonymous) @ FetchQueueWorker.ts:54
t.__tryOrUnsub @ Subscriber.js:192
t.next @ Subscriber.js:130
t._next @ Subscriber.js:76
t.next @ Subscriber.js:53
t.notifyNext @ mergeMap.js:85
t._next @ innerSubscribe.js:14
t.next @ Subscriber.js:53
(anonymous) @ subscribeToArray.js:5
c @ innerSubscribe.js:91
t._innerSub @ mergeMap.js:72
t._tryNext @ mergeMap.js:66
t._next @ mergeMap.js:49
t.next @ Subscriber.js:53
t._next @ filter.js:38
t.next @ Subscriber.js:53
t.next @ Subject.js:47
publishUpdate @ FetchQueue.ts:76
(anonymous) @ FetchQueue.ts:55
t.__tryOrUnsub @ Subscriber.js:192
t.next @ Subscriber.js:130
t._next @ Subscriber.js:76
t.next @ Subscriber.js:53
t.next @ Subject.js:47
add @ FetchQueue.ts:59
(anonymous) @ backend_srv.ts:90
e._trySubscribe @ Observable.js:43
e.subscribe @ Observable.js:29
e.call @ map.js:18
e.subscribe @ Observable.js:24
(anonymous) @ Observable.js:98
e.toPromise @ Observable.js:96
(anonymous) @ backend_srv.ts:65
c @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
K @ app.1c3a1c85d09a392be724.js:2
i @ app.1c3a1c85d09a392be724.js:2
(anonymous) @ app.1c3a1c85d09a392be724.js:2
(anonymous) @ app.1c3a1c85d09a392be724.js:2
(anonymous) @ backend_srv.ts:30
(anonymous) @ backend_srv.ts:376
c @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
K @ app.1c3a1c85d09a392be724.js:2
i @ app.1c3a1c85d09a392be724.js:2
(anonymous) @ app.1c3a1c85d09a392be724.js:2
(anonymous) @ app.1c3a1c85d09a392be724.js:2
(anonymous) @ backend_srv.ts:30
flush @ PerformanceBackend.ts:41
flush @ Echo.ts:42
Show 65 more frames
app.1c3a1c85d09a392be724.js:2 Uncaught (in promise) {status: 401, statusText: "Unauthorized", data: {…}, config: {…}}config: {method: "POST", url: "api/frontend-metrics", data: {…}, retry: 0, hideFromInspector: true}data: {events: Array(3)}hideFromInspector: truemethod: "POST"retry: 0url: "api/frontend-metrics"__proto__: Objectdata: message: "Unauthorized"__proto__: Objectstatus: 401statusText: "Unauthorized"__proto__: Object
K @ app.1c3a1c85d09a392be724.js:2
c @ app.1c3a1c85d09a392be724.js:2
Promise.then (async)
K @ app.1c3a1c85d09a392be724.js:2
i @ app.1c3a1c85d09a392be724.js:2
(anonymous) @ app.1c3a1c85d09a392be724.js:2
(anonymous) @ app.1c3a1c85d09a392be724.js:2
(anonymous) @ backend_srv.ts:30
flush @ PerformanceBackend.ts:41
flush @ Echo.ts:42
login:1 The keyword 'inner-spin-button' specified to an 'appearance' property is not standardized. It will be removed in the future.
```
</details>

**Environment**:
- Grafana version: 7.5.5
- Data source type & version: N/A
- OS Grafana is installed on: Docker
- User OS & Browser: Chrome 90.0.4430.72
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by torkelo**
Fixes #33764

### CODE REVIEW

File: public/app/app.ts [hunk 1]
```diff
@@ -146,7 +147,9 @@ function initEchoSrv() {
     }
   });
 
-  registerEchoBackend(new PerformanceBackend({}));
+  if (contextSrv.user.orgRole !== '') {
+    registerEchoBackend(new PerformanceBackend({}));
+  }
```
Comment (dprokop) [inline public/app/app.ts:152]: Not sure this fixes the issue @torkelo. The backend still gets registered on the login page and the metrics are sent, yet I don't get why it doesn't throw the Unauthorized.. error now?

Comment (dprokop) [reply public/app/app.ts:152]: ah, and the get role for myself on the login page is `Viewer`. Is this ok?

Comment (torkelo) [reply public/app/app.ts:152]: @dprokop with anonymous access disabled I do not get this backend registered nor any metrics sent

Comment (torkelo) [reply public/app/app.ts:152]: @dprokop do you have anonymous access enabled? if not then that is strange

Comment (dprokop) [reply public/app/app.ts:152]: you are right, its the anonym access.

**Review by dprokop — APPROVED** [review]

**Pull request merged**